### PR TITLE
Enforce Lightning module as source of truth for automatic optimization

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

This is further cleanup from #4317. This PR deletes the train loop's automatic optimization attribute and enforces all accesses go through the lightning module's property. This ensures we don't have divergence across these fields. Since the lightning module is the sole decider for automatic optimization, it looked very odd that the lightning module itself was referring to this property via `self.trainer.train_loop.automatic_optimization` even though `self.automatic_optimization` was already defined. Now that the trainer constructor argument for `automatic_optimization` is gone, we can simplify this. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]
thanks @ananthsub :)

**Review by akihironitta — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Really neat !

File: pytorch_lightning/trainer/optimizers.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def init_optimizers(self, model: LightningModule) -> Tuple[List, List, List]:
                 ' * A list of the previously described dict format, with an optional "frequency" key (int)'
             )
 
-        is_manual_optimization = not self.train_loop.automatic_optimization
+        is_manual_optimization = not model.automatic_optimization
```
Comment (ananthsub) [inline pytorch_lightning/trainer/optimizers.py:83]: @SeanNaren looks like the RPC plugin is passing in a Pipe module here instead of a lightning module, causing this attribute lookup to fail. what do you recommend? changing this to `self.lightning_module.automatic_optimization` instead?

Comment (SeanNaren) [reply pytorch_lightning/trainer/optimizers.py:83]: Yep this is fine, especially considering with #6152 we plan on completely removing this implementation going forth, and recommending users to use FSDP.
